packages/@react-aria/menu/src/useMenuItem.ts

```diff
@@ -182,20 +202,70 @@ export function useMenuItem<T>(props: AriaMenuItemProps, state: TreeState<T>, re
   });
 
   let {pressProps, isPressed} = usePress({onPressStart, onPressUp, isDisabled});
-  let {hoverProps} = useHover({
+  let {hoverProps, isHovered} = useHover({
     isDisabled,
     onHoverStart() {
       if (!isFocusVisible()) {
         state.selectionManager.setFocused(true);
         state.selectionManager.setFocusedKey(key);
+        // focus immediately so that a focus scope opened on hover has the correct restore node
+        let isFocused = key === state.selectionManager.focusedKey;
+        if (isFocused && state.selectionManager.isFocused && document.activeElement !== ref.current) {
+          focusSafely(ref.current);
+        }
+      }
+    },
+    onHoverChange: isHovered => {
+      if (isHovered && isMenuDialogTrigger) {
+        if (!openTimeout.current) {
+          openTimeout.current = setTimeout(() => {
+            openSubMenu.current();
+          }, 500);
```

Comment:
feels a bit long but we might want to consult design

Reply:
i'll shorten for now